pkg/i2gw/providers/common/utils.go

```diff
@@ -226,3 +230,158 @@ func removeBackendRefsDuplicates(backendRefs []gatewayv1.HTTPBackendRef) []gatew
+					RequestHeaderModifier: &gatewayv1.HTTPHeaderFilter{
+						Set:    httpFilter.RequestHeaderModifier.Set,
+						Add:    httpFilter.RequestHeaderModifier.Add,
+						Remove: httpFilter.RequestHeaderModifier.Remove,
+					},
+				}
+				grpcFilters = append(grpcFilters, grpcFilter)
+			}
+
+		case gatewayv1.HTTPRouteFilterResponseHeaderModifier:
+			if httpFilter.ResponseHeaderModifier != nil {
+				grpcFilter := gatewayv1.GRPCRouteFilter{
+					Type: gatewayv1.GRPCRouteFilterResponseHeaderModifier,
+					ResponseHeaderModifier: &gatewayv1.HTTPHeaderFilter{
+						Set:    httpFilter.ResponseHeaderModifier.Set,
+						Add:    httpFilter.ResponseHeaderModifier.Add,
+						Remove: httpFilter.ResponseHeaderModifier.Remove,
+					},
+				}
+				grpcFilters = append(grpcFilters, grpcFilter)
+			}
+
+		// These HTTP filter types are not applicable to gRPC
+		case gatewayv1.HTTPRouteFilterRequestRedirect:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		case gatewayv1.HTTPRouteFilterURLRewrite:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		case gatewayv1.HTTPRouteFilterRequestMirror:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		case gatewayv1.HTTPRouteFilterExtensionRef:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		default:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		}
+	}
+
+	// Ensure we return empty slices instead of nil
+	if grpcFilters == nil {
+		grpcFilters = []gatewayv1.GRPCRouteFilter{}
+	}
+	if unsupportedTypes == nil {
+		unsupportedTypes = []gatewayv1.HTTPRouteFilterType{}
+	}
+	
+	return GRPCFilterConversionResult{
+		GRPCFilters:      grpcFilters,
+		UnsupportedTypes: unsupportedTypes,
+	}
+}
+
+// RemoveGRPCRulesFromHTTPRoute removes HTTPRoute rules that target gRPC services
+func RemoveGRPCRulesFromHTTPRoute(httpRoute *gatewayv1.HTTPRoute, grpcServiceSet map[string]struct{}) []gatewayv1.HTTPRouteRule {
+	var remainingRules []gatewayv1.HTTPRouteRule
+
+	for _, rule := range httpRoute.Spec.Rules {
+		var remainingBackendRefs []gatewayv1.HTTPBackendRef
+
+		// Check each backend ref in the rule
+		for _, backendRef := range rule.BackendRefs {
+			serviceName := string(backendRef.BackendRef.BackendObjectReference.Name)
+			// Only keep backend refs that are NOT gRPC services
+			if _, isGRPCService := grpcServiceSet[serviceName]; !isGRPCService {
+				remainingBackendRefs = append(remainingBackendRefs, backendRef)
+			}
+		}
+
+		// If any backend refs remain, keep the rule
+		if len(remainingBackendRefs) > 0 {
+			rule.BackendRefs = remainingBackendRefs
+			remainingRules = append(remainingRules, rule)
+		}
+	}
+
+	return remainingRules
+}
+
+// CreateBackendTLSPolicy creates a BackendTLSPolicy for the given service
+func CreateBackendTLSPolicy(namespace, ingressName, serviceName string) gatewayv1alpha3.BackendTLSPolicy {
+	policyName := fmt.Sprintf("%s-%s-backend-tls", ingressName, serviceName)
```

Comment:
maybe support policyName as a param as well, as this will be used by other providers and who knows what they would want to support